### DISCUSSION TITLE
ReadMe.md errata

### DISCUSSION THREAD

**Discussion by LeeTibbert** [armanbilge/epollcat#46] [General]
First, thank you for the `example` directory.  I was going to create an Issue today asking for such.

Now, on to a few errata I noticed in `ReadMe.md`.  My goal is to be helpful, not critical
in the bad sense.

1) `It currently only supports Linux.`  You picked up describing recent, hard won, macOS support
    in a couple of other places but not here. 
    Later: The macOS support should probably mention both Intel and Apple Mx/M1 architectures
    are supported.  M1 tends to be an odd duck.

2)  Where you would normally use IOApp or IOApp.Simple, simply use EpollApp or EpollApp.Simple 
     instead: it's a drop-in replacement.

    I hate to confess, but I spent most of yesterday, Saturday, trying to use that instruction and
    the Cats IO introduction, tutorial, and .g8 template to create a working epollcat app and failed.

    I suggest at the very least the magic include be given (and probably instructions to remove
    the cats IOApp include).   The next step would be to mention the LibraryInclude or 
    give a reference to it. I think there is something mentioned somewhere else
    about it, all in one place (a note at the bottom of the ReadMe.md so as to
    not distract flow of ideas).

    I got lost in the perpetual sbt "%" vs "%%" vs "%%%" Hell.  Figured it out
    eventually.

    I'll try again once I get some monkeys off my back.

    The `example` will help a lot, but it is not `cut & paste`. Understandably,
    it is tightly integrated with the `core` just built.  You might want to mention that 
    in the example file (and reference a 'cut & paste' HelloWorld with the 
    include that a user would use.)

**Comment by armanbilge** [comment]
Thanks, I pushed a commit fixing the README, indeed I had missed that one :)

You are right, the current README and example is not particularly helpful for someone looking to get started. We can definitely do a bit better here, but we are limited by the fact that this is an auxiliary project: I don't expect it to be used standalone.

Most users will probably want to use http4s, skunk, fs2-io on Scala Native, and will need this project to do so.

So my proposal is that once those projects publish their first Scala Native releases (hopefully in the next week or two), we add examples and possibly even create a g8 template that demonstrate how to use those projects on Scala Native, in which epollcat will play a key part.

There are many interesting examples that we can adapt for Scala Native:
- https://github.com/typelevel/fs2-chat
- https://github.com/http4s/http4s-io.g8
- https://github.com/djspiewak/kitteh-redis
- https://github.com/armanbilge/http4s-cli

I believe those are far more interesting and illustrative than anything we can do here. How does that sound?

**Reply by LeeTibbert** [reply]
Sounds like a plan.  

I will read the current ReadMe.md  as I can.
